roboteam_ai/src/world/Field.cpp

```diff
@@ -38,23 +38,20 @@ Vector2 Field::get_their_goal_center() {
     return Vector2(field.field_length/2, 0);
 }
 
-bool Field::pointIsInDefenceArea(const Vector2 &point, bool isOurDefenceArea, float margin, bool includeOutsideField) {
+bool Field::pointIsInDefenceArea(const Vector2 &point, bool isOurDefenceArea, double margin, bool includeOutsideField) {
     auto defenseArea = getDefenseArea(isOurDefenceArea, margin, includeOutsideField);
     return defenseArea.contains(point);
 }
 
 // the margin is pointed inside the field!
-bool Field::pointIsInField(const Vector2 &point, float margin) {
-    roboteam_msgs::GeometryFieldSize _field = get_field();
-
-    float halfLength = _field.field_length*0.5f;
-    float halfWidth = _field.field_width*0.5f;
+bool Field::pointIsInField(const Vector2 &point, double margin) {
+    auto field = get_field();
 
-    return (point.x <= halfLength - margin &&
-            point.x >= - halfLength + margin &&
-            point.y <= halfWidth - margin &&
-            point.y >= - halfWidth + margin);
+    double halfLength = field.field_length*0.5f - margin;
```

Comment:
the margin is now divided by two since it is only applied once! This thus looks misleading... The test does fail as well on the (literal) edge cases.